### ISSUE TITLE
Incorrect type annotation for service_id parameter to botocore.signers.RequestSigner

### ISSUE THREAD

**Issue by nint8835** [youtype/botocore-stubs#2]
Follow-up issue to https://github.com/youtype/mypy_boto3_builder/issues/195 upon a bit of further testing, looks like I'm hitting an issue that's one part PyCharm issue, one part botocore-stubs issue.

In botocore's [docstrings](https://github.com/boto/botocore/blob/develop/botocore/signers.py#L47-L48), it says that `RequestSigner` takes a `botocore.model.ServiceId`, however in botocore-stubs, it says it takes a`str`.

# Repro

- `pip install boto3-stubs[sts]`

```python
import boto3
from botocore.signers import RequestSigner

session = boto3.session.Session(aws_access_key_id="test", aws_secret_access_key="test")
client = session.client("sts", region_name="ca-central-1")
service_id = client.meta.service_model.service_id

invalid_signer = RequestSigner(
    "sts",
    "ca-central-1",
    "sts",
    "v4",
    session.get_credentials(),
    session.events,
)

valid_signer = RequestSigner(
    service_id,
    "ca-central-1",
    "sts",
    "v4",
    session.get_credentials(),
    session.events,
)

invalid_signer.generate_presigned_url(
    {"method": "GET", "url": "", "body": {}, "headers": {}, "context": {}},
    region_name="ca-central-1",
    expires_in=60,
    operation_name="",
)

valid_signer.generate_presigned_url(
    {"method": "GET", "url": "", "body": {}, "headers": {}, "context": {}},
    region_name="ca-central-1",
    expires_in=60,
    operation_name="",
)
```

## Expected result

- The `invalid_signer` should either result in a mypy error, or should work when you call `get_presigned_url`
- The `valid_signer` should either result in no mypy error, or should not work when you call `get_presigned_url`

## Actual result

- The call to `generate_presigned_url` on `invalid_signer` results in an attribute error due to attempting to access `hyphenize` on `str`, whereas it only exists on `ServiceId`, and does not result in a mypy error
- The call to `generate_presigned_url` on `valid_signer` works, but gives a type error in PyCharm. This part is partially the PyCharm issue, as I'm pretty sure it's a false positive in PyCharm due to ServiceId that we're passing being a subclass of `str`, so even with the current type annotations I believe this should be valid.
<img width="531" alt="image" src="https://user-images.githubusercontent.com/1741548/229121973-b0ee8e08-cb73-4e47-a15e-9f30571c94f4.png">

**Comment by vemel** [comment]
Thank you for this well-written bug report!

I fixed `service_id: ServiceId` argument type. The fix is included in `botocore-stubs 1.29.103.post1`. Please test it and let me know if it works as expected. Regarding `PyCharm` issue - I will check if it works as well.

**Comment by nint8835** [comment]
Tested and confirmed the mypy issue is resolved, thanks once again for the quick fix!

With regards to the PyCharm issue, I'm still running into it but a bit differently now - I'm getting
> Expected type 'ServiceId | ServiceId', got '() -> ServiceId | CachedProperty' instead
When trying to pass `session.client(...).meta.service_model.service_id` as the `service_id` argument. My best guess is it's probably something on the go with the `CachedProperty` decorator on the service_id in the service_model. I know PyCharm doesn't properly support some recent mypy features, that might be what's on the go there. I can open a separate issue for that problem if you want, I'm also content to just leave it as it is where it seems to be an editor-specific issue.
